### DISCUSSION TITLE
426 (Upgrade Required)

### DISCUSSION THREAD

**Discussion by 4ntibala** [obs-websocket-community-projects/obs-websocket-js#248] [Q&A]
Hi,

i added the repo to my angular2 project. After a bit of hassle with the npm packages i got it to compile without errors. 

Sadly, when i try to connect to obs-studio i get 2 essential errors. The first one is Chrome complaining about `Access-Control-Allow-Origin`.. which i managed to get around by using a CORS browser extension.

The second, and main issue is, that even after passing the CORS issue i run into this error from in Chrome inspector:

```zone-evergreen.js:1042 GET http://localhost:4444/ 426 (Upgrade Required)```
```obs.service.ts:38 {status: "error", description: "Connection error.", code: "CONNECTION_ERROR", error: "Connection error."}```

I am quite clueless what to do now. What i want to achieve is connecting from my Angular2 frontend to the obs-websocket-js plugin. I know it was quite tricky to get the repo compiled in Angular2, but the error does not seem to be angular-related (at least not to me).

Is there somebody who can help me out here? Or who had a similar issue? 

I am working from Linux Mint 19, using OBS Studio 25.0.3 with the latest obs-websocket-js & nodejs v8.17.0.

Any help is greatly appreciated! Thanks! :)

**Comment by geo242** [comment]
Is the browser even trying to connect to localhost:4444? Is there activity in the network tab in dev tools? Is the `Upgrade Required` message being returned from network activity or is it something internal to your javascript app?
